assets/src/scripts/export.js

```diff
@@ -50,22 +85,88 @@ jQuery( function ( $ ) {
-				if ( typeof val === 'boolean' ) {
-					v = val;
-				} else {
-					v = val === 'true';
+				// Initialize checkboxes from cookie
+				let was_checked = 0;
+				if ( json_cookie.hasOwnProperty( name ) ) {
+					was_checked = json_cookie[ name ];
 				}
-				$( this ).prop( 'checked', v );
+				$( this ).prop( 'checked', !! was_checked );
 			}
+			// If there's a dependency error, then forcibly uncheck
 			if ( $( this ).attr( 'disabled' ) ) {
 				$( this ).prop( 'checked', false );
 			}
 		} )
 		.change( function () {
-			Cookies.set( 'pb_' + $( this ).attr( 'name' ), $( this ).prop( 'checked' ), {
-				path: '/',
-				expires: 365,
+			let name = $( this ).attr( 'name' );
+			if ( $( this ).prop( 'checked' ) ) {
+				// Cookie syntax: 'ef[<format>]': 1
+				// I.e: 'ef[print_pdf]': 1
+				json_cookie[ name ] = 1;
+			} else {
+				delete json_cookie[ name ];
+			}
+			update_json_cookie();
+		} );
+
+	/* Pins */
+	$( 'td.column-pin' )
+		.find( 'input' )
+		.each( function () {
+			if ( $( this ).prop( 'checked' ) ) {
+				let tr = $( this ).closest( 'tr' );
+				let id = tr.attr( 'data-id' );
+				let cb = $( `input[name='ID[]'][value='${id}']` );
+				$( this ).prop( 'checked', true );
+				cb.prop( 'checked', false );
+				cb.prop( 'disabled', true );
+				tr.find( 'td.column-file span.delete' ).hide();
+			}
+		} )
+		.change( function () {
+			let name =  $( this ).attr( 'name' );
+			let tr = $( this ).closest( 'tr' );
+			let id = tr.attr( 'data-id' );
+			let cb = $( `input[name='ID[]'][value='${id}']` );
+			let format = tr.attr( 'data-format' );
+			if ( $( this ).prop( 'checked' ) ) {
+				_pb_export_pins_inventory[ name ] = format;
+				// Up to five files can be pinned at once.
+				if ( Object.keys( _pb_export_pins_inventory ).length > 5 ) {
+					delete _pb_export_pins_inventory[ name ];
+					$( this ).prop( 'checked', false );
+					alert( PB_ExportToken.maximumFilesWarning );
+					return false;
+				}
+				// If the user has pinned three files of a given export type and they then try to pin an additional file of that type,
+				// an error should be displayed instructing them to deselect one of the pinned files before attempting to pin another.
+				if ( Object.values( _pb_export_pins_inventory ).filter( function ( value ) {
+					// value matches <crc32-format-td>
+					return value === format;
+				} ).length > 3 ) {
+					delete _pb_export_pins_inventory[ name ];
+					$( this ).prop( 'checked', false );
+					alert( PB_ExportToken.maximumFileTypeWarning );
+					return false;
+				}
+				// Checked
+				cb.prop( 'checked', false );
+				cb.prop( 'disabled', true );
+				tr.find( 'td.column-file span.delete' ).hide();
+			} else {
+				// Unchecked
+				delete _pb_export_pins_inventory[ name ];
+				cb.prop( 'disabled', false );
+				tr.find( 'td.column-file span.delete' ).show();
+			}
+			$.ajax( {
+				url: ajaxurl,
+				type: 'POST',
+				data: {
+					action: 'pb_update_pins',
```

Comment:
We should display a success notice (which fades out) for successful pinning. It should also function as an aria-live region to announce the successful event. Something like:

```php
<div class="notice notice-success fade">
<?php
  printf(
    __( 'The file %s has been pinned successfully.', 'pressbooks' ),
    $filename
  );
?>
</div>
```

Reply:
Code is in place but I am having a hard time getting `notice notice-success fade` to not feel awkward. It makes the interface jump around while pinning. Maybe this could be screen-reader only? Please tweak. Currently has no classes...

Reply:
I am changing it to `.screen-reader-only`.